LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Operations/OperationPreferredDateCalculator.swift

```diff
@@ -84,14 +86,22 @@ struct OperationPreferredDateCalculator {
         }
     }
 
-    // If the time elapsed since the last profile removal exceeds the current date plus maintenance period (expired), we should proceed with scheduling a new opt-out request as the broker has failed to honor the previous one.
+    // If the time elapsed since the last profile removal exceeds the current date plus maintenance period (expired),
+    // and the number of attempts is still fewer than the configurable limit,
+    // we should proceed with scheduling a new opt-out request as the broker has failed to honor the previous one.
     private func shouldScheduleNewOptOut(events: [HistoryEvent],
                                          extractedProfileId: Int64,
-                                         schedulingConfig: DataBrokerScheduleConfig) -> Bool {
+                                         schedulingConfig: DataBrokerScheduleConfig,
+                                         attemptCount: Int64?) -> Bool {
         guard let lastRemovalEvent = events.last(where: { $0.type == .optOutRequested && $0.extractedProfileId == extractedProfileId }) else {
             return false
         }
 
+        let currentAttempt = attemptCount ?? 0
+        if schedulingConfig.maxAttempts != -1, currentAttempt >= schedulingConfig.maxAttempts {
+            return false
+        }
+
```

Comment:
We could re-order the below as follows to improve performance:

```suggestion
  let currentAttempt = attemptCount ?? 0
        if schedulingConfig.maxAttempts != -1, currentAttempt >= schedulingConfig.maxAttempts {
            return false
        }

        guard let lastRemovalEvent = events.last(where: { $0.type == .optOutRequested && $0.extractedProfileId == extractedProfileId }) else {
            return false
        }
```